src/appleseed-max-impl/appleseedrenderelement/appleseedrenderelement.cpp

```diff
@@ -76,21 +78,20 @@ namespace
     
     std::vector<std::string> get_aov_names()
     {
-        const asr::AOVFactoryRegistrar factory_registrar;
-        auto factories = factory_registrar.get_factories();
-        auto aov_names = std::vector<std::string>();
-        for (size_t i = 0, e = factories.size(); i < e; ++i)
-            aov_names.push_back(factories[i]->get_model_metadata().get("label"));
+        auto factories = asf::array_vector<std::vector<asr::IAOVFactory*>>(g_aov_factory_registrar.get_factories());
```

Comment:
> Seems that FactoryArrayType can't be used with range loops, it says error C3312: no callable 'begin' function found for type 'renderer::AOVFactoryRegistrar::FactoryArrayType'

Oh ok, the code was fine then, sorry.

> Should I do auto factories = asf::array_vector<std::vector<asr::IAOVFactory*>>(g_aov_factory_registrar.get_factories()); ?

No please, that's a lot worse with allocations, constructions, destructions, copying... :)